### ISSUE TITLE
Ball too small if using anti-stretch and y scale below 100%.

### ISSUE THREAD

**Issue by Crrispy** [vpinball/vpinball#1197]
When checking the anti-stretch ball in the VPX settings, the ball is round , but instead of counter-stretching the "compressed" axis, it compresses the "normal" one. The result is a way too small ball which does not match "reality".

Adjust a table to something around 100% x scale and 85% y scale (about standard to fit a 16:9 screen), with anti-stretch ball not activated. The ball has the proper width but height follows the y scale, so it's not round anymore.
Then activate ball anti-stretch and launch the table again. Instead of being 100/100%, the ball has become 85/85% in size, or even less, it really looks too small.

**Expected behavior**
Ball should be round but at the "proper" size.

**Screenshots**
![20231125_171716](https://github.com/vpinball/vpinball/assets/28100190/b94cf996-3ef8-4ac4-af24-661f23ddd17b)

**Versions & Settings**
 - VPX: 10.8 build 1806 DX x64
 - Windows 11
 - RTX 4070Ti driver 546

**Comment by vbousquet** [comment]
There won't be any solution for an unstretched ball to look goods with regards to its urrounding. It will either overlay over it if enlarged or the oppposite like you show.

**Comment by Crrispy** [comment]
> There won't be any solution for an unstretched ball to look goods with regards to its urrounding. It will either overlay over it if enlarged or the oppposite like you show.

But I didn't have this issue in older builds.

**Comment by manuelwehrli** [comment]
Ball anti-stretch worked perfectly in 10.7.x, even in 9.x it was possible with the right value manually set. So there must be a way to achieve it. It looks like some axis is inverted in the calculation, because the ball gets bigger in the upper part of the table. But regarding perspective, it should be the opposite, bigger in the lower part where it is closer to the player and smaller in the upper part where it is more far away from the player.

**Comment by michroch** [comment]
> > There won't be any solution for an unstretched ball to look goods with regards to its urrounding. It will either overlay over it if enlarged or the oppposite like you show.
> 
> But I didn't have this issue in older builds. Would it be possible to add a "ball scale" setting ?

Totally agree with Crrispy, I went back to build 1795 before this change cause the ball is so weird looking now in the latest build

**Comment by Crrispy** [comment]
More testing:
1) I set the table to 100% all scales. The ball is a bit bigger but the base issue is still there, the ball is smaller on the bottom of the playfield than at the top. 
2) I then reverted to build 1798, normal behaviour and ball size.
All of this with "Force round ball" checked.

I think the bug tag should be restored instead of "enhancement"....

**Comment by Meshuggah70** [comment]
I too have had this issue since the "new anti-stretch"  was implemented also on version  1826 the ball went from marble size when anti stretch is checked to completely invisible

**Comment by keithhov** [comment]
I agree with above, ball size issues present since force round ball introduced, when force round ball in uncheck, the ball is flattened, either way.  Maybe the ball should be bigger at the top of the table, since its higher as the top of the table is higher, but it's also farther away from the player.  Minimally it needs to be toned way down.  
![Screenshot 2023-11-28 221312](https://github.com/vpinball/vpinball/assets/146595499/f1c24b4c-6e98-4e34-a92e-1e33e19b754f)

**Comment by timdeporte** [comment]
In the latest builds it is indeed impossible for cabinet users to anti-stretch the ball. It always looks like I am hitting an egg. And like previous users mentioned in this thread, using the new anti-stretch option makes the ball look like a tiny marble at the bottom of the screen and normal size in the upper part of the screen. Should be switched. I am using older builds unfortunately.

**Comment by timdeporte** [comment]
A workaround is to create a pov-file in the older builds and then use the ‘Legacy Mode’ in the latest build. This mode will still read the pov-file instead of the newer ini-file. But the new ‘Window Mode’ still needs some fixing imo to make full use of the new pov functionality for cabinet users.

**Comment by toxieainc** [comment]
@vbousquet I would consider this also a blocker for 10.8, or what do you think? I'd like to avoid instant legacy.. Any alternatives you could think of?

**Comment by vbousquet** [comment]
I'm not sure there are any clean alternative. The current code ensure a round ball (which is somewhat wrong with respect to perspective) and ensure it will not overwrite other parts by only shrinking. I initially tried the other way round but people complained about the ball being too big. Perhaps a compromise would be to use an average to half shrink on one axis, half enlarge on the other.

**Comment by timdeporte** [comment]
@vbousquet I understand you want to respect perspective for cabinet users but I don’t fully understand why the ball needs to shrink/enlarge in width. Cabinet users already have a 3-dimensional setting. I agree you want to change the height of the ball depending on its position on the playfield to respect perspective. But the width of the ball does not need to change because I keep standing in front of my cabinet. The new anti-stretch option still stretches the ball in all dimensions on a ‘static’ playfield. The ball size is way off when it reaches the flippers. The thing is that the new option does not keep the dimensions of the ball fixed anymore. Because cabinet users tend to set their X axis higher for each table (increase the width of the table) they need to anti-stretch the ball to undo the oval shape of the ball (too wide).

**Comment by Crrispy** [comment]
Why not reproduce the same thing as in current 10.8.0 ? I don't use 10.8.1 builds mainly due to this issue.

**Comment by Crrispy** [comment]
At which build was this change implemented exactly? I <think> this issue should revert to the "bug" label, it's not normal that the ball is so small at the bottom and bigger at the top when force round ball is enabled. I might try older 10.8.0 builds to try to guess when it happened.

**Comment by jsm174** [comment]
@Crrispy, I think you should be able to figure it out from the commit history.

You'd want to grab an action build immediately before this one: https://github.com/vpinball/vpinball/commit/6bb53749ab7fe0b72bf9046ebb0c96a7afa47045

See this issue as well: https://github.com/vpinball/vpinball/issues/1171

**Comment by Crrispy** [comment]
Thanks, I'll try this build to see how it behaves. There should be something possible in between extremes I suppose. But I'm pretty sure there will be more complaints on this when more useres start using newer builds.

**Comment by timdeporte** [comment]
Merry Christmas to you all! 
Would it be possible to leave the ‘(compensate perspective stretch)’ as an extra option? Just to let the user decide how to handle the compensation. To be honest I don’t see a lot of extra benefit in using the perspective compensation as the roundness of the ball is not distorted a lot when I play. The playfield screen is already physically tilted and the angle between my eyes and the top of the screen don’t actually need any compensation. At least for an average length adult. The perspective compensation is currently too extreme.

**Comment by Crrispy** [comment]
I totally second this, or make a slider for it to choose compensation amplitude.

**Comment by keithhov** [comment]
Much better looking ball with 1850... not perfect but much better.  I have a cabinet setup, I have to uncheck force round ball to get a ball close to round,  tested on 3 or 4 tables,  I had to narrow some tables a bit to get a round ball.

**Comment by Crrispy** [comment]
Hello, thanks for this. I've opened a new one unfortunately, #1322.
